crates/net/src/sim2h_worker.rs

```diff
@@ -389,24 +355,25 @@ impl NetWorker for Sim2hWorker {
 
     /// Check for messages from our NetworkEngine
     fn tick(&mut self) -> NetResult<bool> {
-        if Instant::now().duration_since(self.time_of_last_sent) > Duration::from_secs(PING_DURATION_SECS) {
-            self.send_ping();
-        }
-        if let Err(transport_error) = detach_run!(&mut self.transport, |t| t.process(self)) {
-            error!("Transport error: {:?}", transport_error);
-            // This most likely means we have connection issues.
-            // Send ping to reestablish a potentially lost connection.
-            self.send_ping();
-        }
-        let mut did_something = WorkWasDone::from(false);
+        let mut did_something = false;
 
-        let client_messages = self.inbox.drain(..).collect::<Vec<_>>();
-        for data in client_messages {
-            debug!("CORE >> Sim2h: {:?}", data);
-            if let Err(error) = self.handle_client_message(data) {
-                error!("Error handling client message in Sim2hWorker: {:?}", error);
-            }
-            did_something = WorkWasDone::from(true);
+        match self.stream_manager.connection_status(&self.server_url.clone().into()) {
+            ConnectionStatus::None => {
+                warn!("No connection to sim2h server. Trying to reconnect...");
+                self.stream_manager.connect(&self.server_url.clone().into())?;
+            },
+            ConnectionStatus::Initializing => debug!("connecting..."),
+            ConnectionStatus::Ready => {
+                let client_messages = self.inbox.drain(..).collect::<Vec<_>>();
```

Comment:
do we have to collect here or can we just use a for_each() to get the same results on the iterator? Just asking cause we are collecting and then iterating again

Reply:
I thought same thing but you end up mutably borrowing `self` twice using that pattern and the compiler complains. Must be some way to do that without a full copy of the collections though.